src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -346,27 +374,36 @@ private void ConfigureJsonTextReader(JsonTextReader reader)
         /// <param name="jsonRpcMessage">A JSON-RPC message.</param>
         private void TokenizeUserData(JsonRpcMessage jsonRpcMessage)
         {
-            if (jsonRpcMessage is Protocol.JsonRpcRequest request)
+            try
             {
-                if (request.ArgumentsList != null)
-                {
-                    request.ArgumentsList = request.ArgumentsList.Select(this.TokenizeUserData).ToArray();
-                }
-                else if (request.Arguments != null)
+                if (jsonRpcMessage is Protocol.JsonRpcRequest request)
                 {
-                    if (this.ProtocolVersion.Major < 2)
+                    this.requestIdBeingSerialized = Convert.ToInt64(request.Id);
```

Comment:
>Convert.ToInt64(request.Id [](start = 52, length = 26)

What happens if `request.Id` is null (for outbound notifications)? Will `Convert.ToInt64` throw since there isn't a `long` that can be based on that? If it is `null`, we can set `this.requestIdBeingSerialized` to null.

Reply:
Convert.ToInt64 will actually return 0 if request.Id is null. I did a quick test to confirm it.

---
In reply to: [312682197](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r312682197) [](ancestors = 312682197)

Reply:
We don't want it to be 0. If it is, the code won't throw an exception when `Progress<T>` is passed in an notification (one without an ID). We need the field to be set to `null` in that case.